From 86f628b03c0f085a095b021918a14c8ad99b79e9 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:48:23 -0400
Subject: [x86] perf: Fix IVT/SNB-EP uncore CBOX NID filter table

Message-id: <1408953210-26343-16-git-send-email-jolsa@redhat.com>
Patchwork-id: 88730
O-Subject: [PATCH RHEL7.1 BZ1133083 015/322] perf/x86/uncore: Fix IVT/SNB-EP uncore CBOX NID filter table
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 337397f3afc911d94d1d71371a36a53ce218b41f
Author: Stephane Eranian <eranian@google.com>
Date: Wed Feb 19 14:10:18 2014 +0100

description
===========
This patch updates the CBOX PMU filters mapping tables for SNB-EP
and IVT (model 45 and 62 respectively).

The NID umask always comes in addition to another umask.
When set, the NID filter is applied.

The current mapping tables were missing some code/umask
combinations to account for the NID umask. This patch
fixes that.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/cpu/perf_event_intel_uncore.c b/arch/x86/kernel/cpu/perf_event_intel_uncore.c
index b6ecb1f..25b48dc 100644
--- a/arch/x86/kernel/cpu/perf_event_intel_uncore.c
+++ b/arch/x86/kernel/cpu/perf_event_intel_uncore.c
@@ -460,8 +460,11 @@ static struct extra_reg snbep_uncore_cbox_extra_regs[] = {
  SNBEP_CBO_EVENT_EXTRA_REG(SNBEP_CBO_PMON_CTL_TID_EN,
       SNBEP_CBO_PMON_CTL_TID_EN, 0x1),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0334, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4334, 0xffff, 0x6),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0534, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4534, 0xffff, 0x6),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0934, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4934, 0xffff, 0x6),
  SNBEP_CBO_EVENT_EXTRA_REG(0x4134, 0xffff, 0x6),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0135, 0xffff, 0x8),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0335, 0xffff, 0x8),
@@ -1039,10 +1042,15 @@ static struct extra_reg ivt_uncore_cbox_extra_regs[] = {
  SNBEP_CBO_EVENT_EXTRA_REG(SNBEP_CBO_PMON_CTL_TID_EN,
       SNBEP_CBO_PMON_CTL_TID_EN, 0x1),
  SNBEP_CBO_EVENT_EXTRA_REG(0x1031, 0x10ff, 0x2),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x1134, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4134, 0xffff, 0xc),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x5134, 0xffff, 0xc),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0334, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4334, 0xffff, 0xc),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0534, 0xffff, 0x4),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4534, 0xffff, 0xc),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0934, 0xffff, 0x4),
- SNBEP_CBO_EVENT_EXTRA_REG(0x4134, 0xffff, 0xc),
+ SNBEP_CBO_EVENT_EXTRA_REG(0x4934, 0xffff, 0xc),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0135, 0xffff, 0x10),
  SNBEP_CBO_EVENT_EXTRA_REG(0x0335, 0xffff, 0x10),
  SNBEP_CBO_EVENT_EXTRA_REG(0x2135, 0xffff, 0x10),
-- 
1.7.1